src/app/screens/speedUpTransaction/index.tsx

```diff
@@ -0,0 +1,541 @@
+import BigNumber from 'bignumber.js';
+import { useEffect, useState } from 'react';
+import toast from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import { NumericFormat } from 'react-number-format';
+import { useNavigate, useParams } from 'react-router-dom';
+import { MoonLoader } from 'react-spinners';
+import { useTheme } from 'styled-components';
+import CustomFee from './customFee';
+import {
+  ButtonContainer,
+  Container,
+  ControlsContainer,
+  CustomFeeIcon,
+  DetailText,
+  FeeButton,
+  FeeButtonLeft,
+  FeeButtonRight,
+  HighlightedText,
+  LoaderContainer,
+  SecondaryText,
+  StyledActionButton,
+  SuccessActionsContainer,
+  Title,
+  WarningText,
+} from './index.styled';
+
+type TierFees = {
+  enoughFunds: boolean;
+  fee?: number;
+  feeRate: number;
+};
+
+type RbfRecommendedFees = {
+  medium?: TierFees;
+  high?: TierFees;
+  higher?: TierFees;
+  highest?: TierFees;
+};
+
+function SpeedUpTransactionScreen() {
+  const { t } = useTranslation('translation', { keyPrefix: 'SPEED_UP_TRANSACTION' });
+  const theme = useTheme();
+  const navigate = useNavigate();
+  const [showCustomFee, setShowCustomFee] = useState(false);
+  const { selectedAccount, accountType, network, btcFiatRate, fiatCurrency } = useWalletSelector();
+  const seedVault = useSeedVault();
+  const { id } = useParams();
+  const btcClient = useBtcClient();
+  const [rbfTxSummary, setRbfTxSummary] = useState<{
+    currentFee: number;
+    currentFeeRate: number;
+    minimumRbfFee: number;
+    minimumRbfFeeRate: number;
+  }>();
+  const [feeRateInput, setFeeRateInput] = useState<string | undefined>();
+  const [totalFee, setTotalFee] = useState<string | undefined>();
+  const [selectedOption, setSelectedOption] = useState<string | undefined>();
+  const [recommendedFees, setRecommendedFees] = useState<RecommendedFeeResponse>();
+  const [rbfRecommendedFees, setRbfRecommendedFees] = useState<RbfRecommendedFees>();
+  const { data: transaction } = useTransaction(id!);
+  const [rbfTransaction, setRbfTransaction] = useState<any>();
+  const { t: signatureRequestTranslate } = useTranslation('translation', {
+    keyPrefix: 'SIGNATURE_REQUEST',
+  });
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [currentStepIndex, setCurrentStepIndex] = useState(0);
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  const [isConnectSuccess, setIsConnectSuccess] = useState(false);
+  const [isConnectFailed, setIsConnectFailed] = useState(false);
+  const [isTxApproved, setIsTxApproved] = useState(false);
+  const [isTxRejected, setIsTxRejected] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [customFeeRate, setCustomFeeRate] = useState<string | undefined>();
+  const [customTotalFee, setCustomTotalFee] = useState<string | undefined>();
+  const [customFeeError, setCustomFeeError] = useState<string | undefined>();
+
+  const fetchRbfData = async () => {
```

Comment:
Could you put this inside the useEffect at the top? 🙏 Otherwise the deps arg is a bit broken